src/odemis/gui/cont/project.py

```diff
@@ -336,100 +349,174 @@ def _on_text(self, evt):
         """
-        number (int): number of the calibration region
-        tab_data (FastEMAcquisitionGUIData): tab data model
-        view_ctrl (FastEMAcquisitionViewport): viewport controller
+        :param number: (int) The number of the calibration region.
+        :param tab_data: (FastEMAcquisitionGUIData) The tab data model.
+        :param view_ctrl (FastEMAcquisitionViewport) The viewport controller. TODO replace with view only.
         """
         self._view_ctrl = view_ctrl
         self._tab_data = tab_data
 
-        # By default, the calibration region is in the center of the scintillator. The size is given by
-        # the ebeam resolution.
-        pos = tab_data.main.scintillator_positions[number]
-        sz = (tab_data.main.ebeam.resolution.value[0] * tab_data.main.ebeam.pixelSize.value[0],
-              tab_data.main.ebeam.resolution.value[1] * tab_data.main.ebeam.pixelSize.value[1])
-
-        l = pos[0] - 0.5 * sz[0]
-        t = pos[1] + 0.5 * sz[1]
-        r = pos[0] + 0.5 * sz[0]
-        b = pos[1] - 0.5 * sz[1]
-
         # Get ROC model (exists already in tab data) and change coordinates
-        self.model = tab_data.calibration_regions.value[number]
-        self.model.coordinates.value = (l, t, r, b)
-        self.model.coordinates.subscribe(self._on_coordinates)
+        self.calib_model = tab_data.calibration_regions.value[number]
+        self.calib_model.coordinates.subscribe(self._on_coordinates)
 
-        # Add overlay
-        self.overlay = view_ctrl.viewports[0].canvas.add_calibration_overlay(self.model.coordinates, number)
+        self.overlay = None
 
     def fit_view_to_bbox(self):
         """
-        Zoom in to calibration region. Calibration region is in the center and takes up 1/100 of viewport area.
+        Zoom into calibration region. Calibration region is centered in the scintillator with the
+        corresponding number and takes up approximately 1/100 of the viewport area.
         """
-        logging.debug("Zooming in to calibration region %s.", self.model.name.value)
+        logging.debug("Zooming into calibration region %s.", self.calib_model.name.value)
         cnvs = self._view_ctrl.viewports[0].canvas
-        l, t, r, b = self.model.coordinates.value
-        sz = (self._tab_data.main.ebeam.resolution.value[0] * self._tab_data.main.ebeam.pixelSize.value[0],
-              self._tab_data.main.ebeam.resolution.value[1] * self._tab_data.main.ebeam.pixelSize.value[1])
-        cnvs.fit_to_bbox([l - 5 * sz[0], t - 5 * sz[1], r + 5 * sz[0], b + 5 * sz[1]])
+        l, t, r, b = self.calib_model.coordinates.value
+        size = (self._tab_data.main.multibeam.resolution.value[0] * self._tab_data.main.multibeam.pixelSize.value[0],
+                self._tab_data.main.multibeam.resolution.value[1] * self._tab_data.main.multibeam.pixelSize.value[1])
+        cnvs.fit_to_bbox([l - 5 * size[0], t - 5 * size[1], r + 5 * size[0], b + 5 * size[1]])
```

Comment:
Why do you do `5*size`?

Reply:
It defines how much we zoom in on the ROC - so that there is still some space shown around the ROC. I added a comment.